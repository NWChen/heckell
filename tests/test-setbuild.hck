let S : int set;
S = {1, 2, 3, 4};
let T : int set;
T = {x in S | x -2  > 0};

let P : int set set;
P = {{1,2}, {2,3}, {5,4}};
let Q : int set set;
Q = {s in P | s < S};

let R : real set;
R = {x in {2.1, 0.3, 4.5} | x == 2.1};

A = { 2*x | x in R, x != 4, x+2, x/2 == 1};
