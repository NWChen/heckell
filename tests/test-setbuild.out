let S: (int set)
S = {1, 2, 3, 4}
let T: (int set)
T = {x in S | (x) ->
  (
    ((x - 2) > 0)
  )}
let P: ((int set) set)
P = {{1, 2}, {2, 3}, {5, 4}}
let Q: ((int set) set)
Q = {s in P | (s) ->
  (
    (s < S)
  )}
let R: (real set)
R = {x in {2.1, 0.3, 4.5} | (x) ->
  (
    (x == 2.1)
  )}
A = {(x) ->
  (
    (2 * x)
  ) | x in R, (x) ->
  (
    (((x != 4) and (x + 2)) and ((x / 2) == 1))
  )}
B = {(x) ->
  (
    (x + 2)
  ) | x in R, (x) ->
  (
    true
  )}
